In [ ]:
!pip install ydata-profiling >> dev/null

/bin/bash: line 1: dev/null: No such file or directory


In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from ydata_profiling import ProfileReport

In [4]:
df = pd.read_csv('datasets/EDUCATION_ATTAINMENT.csv')
df.head()

Ages[15-49]_All_grade1  Ages[15-49]_All_grade2  Ages[15-49]_All_grade3  \
0                0.669091                0.667214                0.651812   
1                0.485879                0.480171                0.465510   
2                0.516431                0.514233                0.502269   
3                0.914294                0.903528                0.871690   
4                0.876990                0.841497                0.762864   

   Ages[20-29]_All_grade1  Ages[20-29]_All_grade2  Ages[20-29]_All_grade3  \
0                0.460899                0.458749                0.446925   
1                0.267236                0.265735                0.257999   
2                0.327775                0.326250                0.318003   
3                0.853770                0.840792                0.810794   
4                0.881265                0.854158                0.795892   

   Ages[30-39]_All_grade1  Ages[30-39]_All_grade2  Ages[30-39]_All_grade3  \
0                0.292791                0.290929                0.279100   
1                0.191279                0.188839                0.182693   
2                0.221981                0.220472                0.214040   
3                0.775725                0.757109                0.713569   
4                0.883625                0.853812                0.805231   

   Ages[40-49]_All_grade1  ...  Ages[20-29]_Female_grade2  \
0                0.250981  ...                   0.279915   
1                0.187151  ...                   0.122435   
2                0.205611  ...                   0.157744   
3                0.786468  ...                   0.786506   
4                0.747516  ...                   0.793379   

   Ages[20-29]_Female_grade3  Ages[30-39]_Female_grade1  \
0                   0.272185                   0.111428   
1                   0.117916                   0.084555   
2                   0.151130                   0.086072   
3                   0.746073                   0.699617   
4                   0.712726                   0.808248   

   Ages[30-39]_Female_grade2  Ages[30-39]_Female_grade3  \
0                   0.109101                   0.103791   
1                   0.083607                   0.080518   
2                   0.085138                   0.082124   
3                   0.675786                   0.618058   
4                   0.762858                   0.693267   

   Ages[40-49]_Female_grade1  Ages[40-49]_Female_grade2  \
0                   0.098721                   0.097825   
1                   0.065466                   0.065466   
2                   0.073085                   0.073037   
3                   0.687626                   0.648913   
4                   0.559192                   0.516735   

   Ages[40-49]_Female_grade3      country  year  
0                   0.092263  Afghanistan  2015  
1                   0.064012  Afghanistan  2007  
2                   0.071735  Afghanistan  2010  
3                   0.576818       Angola  2015  
4                   0.429649       Angola  1999  

[5 rows x 38 columns]

In [5]:
#Get the generation information about the dataframe
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 626 entries, 0 to 625
Data columns (total 38 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Ages[15-49]_All_grade1     626 non-null    float64
 1   Ages[15-49]_All_grade2     626 non-null    float64
 2   Ages[15-49]_All_grade3     626 non-null    float64
 3   Ages[20-29]_All_grade1     626 non-null    float64
 4   Ages[20-29]_All_grade2     626 non-null    float64
 5   Ages[20-29]_All_grade3     626 non-null    float64
 6   Ages[30-39]_All_grade1     625 non-null    float64
 7   Ages[30-39]_All_grade2     625 non-null    float64
 8   Ages[30-39]_All_grade3     625 non-null    float64
 9   Ages[40-49]_All_grade1     625 non-null    float64
 10  Ages[40-49]_All_grade2     625 non-null    float64
 11  Ages[40-49]_All_grade3     625 non-null    float64
 12  Ages[15-49]_Male_grade1    626 non-null    float64
 13  Ages[15-49]_Male_grade2    626 non-null    float64

In [12]:
#Descriptive analysis of the dataset
df.describe(include='all')

Ages[15-49]_All_grade1  Ages[15-49]_All_grade2  \
count               626.000000              626.000000   
unique                     NaN                     NaN   
top                        NaN                     NaN   
freq                       NaN                     NaN   
mean                  0.873886                0.861490   
std                   0.162624                0.166895   
min                   0.251917                0.244071   
25%                   0.831791                0.803131   
50%                   0.948926                0.935167   
75%                   0.984447                0.980112   
max                   1.000000                1.000000   

        Ages[15-49]_All_grade3  Ages[20-29]_All_grade1  \
count               626.000000              626.000000   
unique                     NaN                     NaN   
top                        NaN                     NaN   
freq                       NaN                     NaN   
mean                  0.837509                0.823849   
std                   0.176555                0.200559   
min                   0.233274                0.079610   
25%                   0.757465                0.744915   
50%                   0.909587                0.910132   
75%                   0.971166                0.973777   
max                   1.000000                1.000000   

        Ages[20-29]_All_grade2  Ages[20-29]_All_grade3  \
count               626.000000              626.000000   
unique                     NaN                     NaN   
top                        NaN                     NaN   
freq                       NaN                     NaN   
mean                  0.812738                0.789965   
std                   0.202587                0.207979   
min                   0.079610                0.079610   
25%                   0.723418                0.678869   
50%                   0.896568                0.864849   
75%                   0.967410                0.955625   
max                   0.999824                0.999264   

        Ages[30-39]_All_grade1  Ages[30-39]_All_grade2  \
count               625.000000              625.000000   
unique                     NaN                     NaN   
top                        NaN                     NaN   
freq                       NaN                     NaN   
mean                  0.766171                0.752519   
std                   0.236563                0.237679   
min                   0.098059                0.091358   
25%                   0.639447                0.612464   
50%                   0.853861                0.831246   
75%                   0.959097                0.948259   
max                   1.000000                1.000000   

        Ages[30-39]_All_grade3  Ages[40-49]_All_grade1  ...  \
count               625.000000              625.000000  ...   
unique                     NaN                     NaN  ...   
top                        NaN                     NaN  ...   
freq                       NaN                     NaN  ...   
mean                  0.724980                0.709166  ...   
std                   0.240681                0.259998  ...   
min                   0.085236                0.036549  ...   
25%                   0.561545                0.527055  ...   
50%                   0.794898                0.785356  ...   
75%                   0.922841                0.937129  ...   
max                   1.000000                1.000000  ...   

        Ages[20-29]_Female_grade2  Ages[20-29]_Female_grade3  \
count                  626.000000                 626.000000   
unique                        NaN                        NaN   
top                           NaN                        NaN   
freq                          NaN                        NaN   
mean                     0.778972                   0.755644   
std                      0.238637                   0.243755   
min            

In [7]:
#Checking the null values
df.isnull().sum()

Ages[15-49]_All_grade1       0
Ages[15-49]_All_grade2       0
Ages[15-49]_All_grade3       0
Ages[20-29]_All_grade1       0
Ages[20-29]_All_grade2       0
Ages[20-29]_All_grade3       0
Ages[30-39]_All_grade1       1
Ages[30-39]_All_grade2       1
Ages[30-39]_All_grade3       1
Ages[40-49]_All_grade1       1
Ages[40-49]_All_grade2       1
Ages[40-49]_All_grade3       1
Ages[15-49]_Male_grade1      0
Ages[15-49]_Male_grade2      0
Ages[15-49]_Male_grade3      0
Ages[20-29]_Male_grade1      0
Ages[20-29]_Male_grade2      0
Ages[20-29]_Male_grade3      0
Ages[30-39]_Male_grade1      1
Ages[30-39]_Male_grade2      1
Ages[30-39]_Male_grade3      1
Ages[40-49]_Male_grade1      1
Ages[40-49]_Male_grade2      1
Ages[40-49]_Male_grade3      1
Ages[15-49]_Female_grade1    0
Ages[15-49]_Female_grade2    0
Ages[15-49]_Female_grade3    0
Ages[20-29]_Female_grade1    0
Ages[20-29]_Female_grade2    0
Ages[20-29]_Female_grade3    0
Ages[30-39]_Female_grade1    1
Ages[30-39]_Female_grade2    1
Ages[30-

In [9]:
profile = ProfileReport(df, title='Education Attainment')
# profile.to_notebook_iframe()

In [11]:
profile.to_file(output_file='profile-report.html')

Export report to file: 100%|██████████████████████| 1/1 [00:00<00:00,  5.31it/s]


### Profile Report Conclusions

Findings from the profile report generated by ydata_profiling:

* There are 38 columns(variables) and 626 rows (observations) in the dataset.
  
* Missing values:
  * There are 18 missing values in total which is 0.1% of the number of values in the dataset.
  * Only 1 value is missing on the columns where missing values are detected

* Duplicates:
     * There is only 1 duplicated row with the country value being Panama in the year 1995  

* Correlations
    * All columns have a positive correlation.
    * The age groups of males between 30-39 have the least coefficient ratio of (0.095) in relation to the year meaning that the grades of males in this age group tend to minimally increase in comparison to other age groups every year.

* Interesting patterns
    * The mean grades of the sample dataset age groups seem to have positively increased from the year 2000 